### PR TITLE
GitBook: Document Derived Keys

### PR DESCRIPTION
**Pull request by AeonSw4n**

### CODE REVIEW

**Review by maebeam — APPROVED** [review]

**Review by lazynina — APPROVED** [review]

**Comment by tijno** [comment]
Awesome!

**Comment by AeonSw4n** [comment]
Don't worry @tijno there's more coming 🙂

**Comment by tijno** [comment]
@AeonSw4n Which ones are in progress and which TODOs are not? Was gonna work on some this week to flesh them out.

**Comment by AeonSw4n** [comment]
Oh that's awesome! There's some deep work in progress for improving identity and backend. I think a really cool TODO that we wanted to have is community tools / starter code documentation page/s. The idea is that a dev could go there and just copy some existing code to immediately get started on their project, as opposed to writing stuff from scratch. I think we could use a hand in there :)

File: devs/identity-api.md [hunk 1]
```diff
@@ -104,6 +104,51 @@ An application should store the current `publicKey` and `users` objects in its l
...
+
+The model requires the owner account to have some balance to execute the `authorizeDerivedKey` transaction. This poses a limitation as the derived key might not be immediately available after promting the user with the identity window. The argument here is that the user doesn't need a derived key until they have some non-zero balance in their account. One possible authorization flow is to send the authorizeDerivedKey transaction right when user has/receives sufficient balance. This simple background mechanism should mitigate most UX issues.
```
Comment (zknerd) [inline devs/identity-api.md:150]: nit: "prompting"

File: devs/backend-api.md [hunk 2]
```diff
@@ -605,6 +605,59 @@ More info on the request [here](https://github.com/deso-protocol/backend/blob/47
 }
 ```
 
+### Authorize Derived Key
+
+```text
+POST /api/v0/authorize-derived-key
+```
+
+Prepare transaction for authorizing a derived key. The payload for this endpoint should be taken from the identity `derive` flow. Transaction needs to be signed and submitted through `api/v0/submit-transaction` before changes come into effect. This transaction can be signed by a "first-time" derived key. Note: before signing any transactions with the derived key, you need to place the derived public key in compressed byte format (33 bytes array) in transaction's `ExtraData["DerivedPublicKey"]`. Endpoint implementation in [backend](https://github.com/deso-protocol/backend/blob/764d71/routes/transaction.go#L2094).
```
Comment (zknerd) [inline devs/backend-api.md:614]: nit: missing front slash in '/api/v0/submit-transaction'

Comment (AeonSw4n) [reply devs/backend-api.md:614]: Thanks! I've just pushed a bigger update to Identity so this PR is obsolete. Will close it now.

**Pull request closed without merging**